core/src/main/java/jenkins/model/ArtifactManager.java

```diff
@@ -60,11 +60,17 @@
      * @param launcher a launcher to use if external processes need to be forked
      * @param listener a way to print messages about progress or problems
      * @param artifacts map from paths in the archive area to paths relative to {@code workspace} (all paths {@code /}-separated)
+     * @param preserveExecute if true, sets the execute bit on the target if the execute bit is set on the source
+     * @param preserveMtime if true, sets the modification time of the target to match the modification time of the source
      * @throws IOException if transfer or copying failed in any way
      * @throws InterruptedException if transfer was interrupted
      * @see ArtifactArchiver#perform(Run, FilePath, Launcher, TaskListener)
      */
-    public abstract void archive(FilePath workspace, Launcher launcher, BuildListener listener, Map<String,String> artifacts) throws IOException, InterruptedException;
+    public abstract void archive(FilePath workspace, Launcher launcher, BuildListener listener,
```

Comment:
As long as artifact archiving is concerned I prefer preserving original attributes by default.